src/main/java/org/sagebionetworks/bridge/hibernate/HibernateAssessmentDao.java

```diff
@@ -109,20 +121,30 @@ final void setHibernateHelper(HibernateHelper hibernateHelper) {
     }
     
     @Override
-    public Optional<Assessment> getAssessment(String appId, String guid) {
+    public Optional<Assessment> getAssessment(String appId, String ownerId, String guid) {
+        QueryBuilder builder = new QueryBuilder();
+        builder.append(GET_BY_GUID, APP_ID, appId, GUID, guid);
+        if (ownerId != null) {
+            builder.append(LIMIT_TO_OWNER, OWNER_ID, ownerId); 
+        }
```

Comment:
But don't we check this after we load it using AuthUtils?

Reply:
We do, but they are not redundant. For example, some calls set the ownerId to null...so they must then check if the caller is a study designer, that they have access to the assessment. They complement each other and eliminate edge cases.

Reply:
We do use both, but they complement each other. In cases where this value is null, we then want to verify with the AuthUtils check.